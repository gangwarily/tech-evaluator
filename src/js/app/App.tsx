import React from 'react';
import { View, Text } from 'react-native';
// Import this for CSS Module (not going to work in React Native)
// import styles from './App.css';

const App = () => (
    <View>
        <Text style={styles.RedText}>
            Hello!
        </Text>
    </View>
);

const styles = {
    RedText: {
        color: 'red',
    },
};

export default App;