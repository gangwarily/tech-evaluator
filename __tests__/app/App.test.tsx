import React from 'react';
import App from '../../src/js/app/App';
import TestRenderer, {ReactTestInstance, ReactTestRenderer} from 'react-test-renderer';

describe('App component unit test', () => {
    it('render test', () => {
        const renderer: ReactTestRenderer = TestRenderer.create(<App />);
        const root: ReactTestInstance = renderer.root;

        expect(root).not.toBeNull();
        expect(renderer.toJSON()).toMatchSnapshot();

        const div: ReactTestInstance = root.findByType('div');
        expect(root.findAllByType('div').length).toEqual(1);

        const span: ReactTestInstance = div.findByType('span');
        expect(root.findAllByType('span').length).toEqual(1);
        expect(span.children[0]).toEqual('Hello!');
    })
});