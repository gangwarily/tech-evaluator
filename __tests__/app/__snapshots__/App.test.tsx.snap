// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component unit test render test 1`] = `
<div>
  <span
    className="RedText"
  >
    Hello!
  </span>
</div>
`;
